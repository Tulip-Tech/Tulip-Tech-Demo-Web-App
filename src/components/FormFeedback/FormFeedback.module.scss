@use 'src/styles/variables';
@use 'src/styles/theme';

.formFeedback {
  margin-bottom: 24px;
  padding: 16px;
  color: variables.$white;
  font-weight: var(--body-font-weight-bold);
  font-size: 18px;
  text-align: left;
  border-radius: 4px;
}

.error {
  color: theme.$form-feedback-error-color;
  background-color: theme.$form-feedback-error-bg-color;
}

.warning {
  color: theme.$form-feedback-warning-color;
  background-color: theme.$form-feedback-warning-bg-color;
}

.success {
  color: theme.$form-feedback-success-color;
  background-color: theme.$form-feedback-success-bg-color;
}
