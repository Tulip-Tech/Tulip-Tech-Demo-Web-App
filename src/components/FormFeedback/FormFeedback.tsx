import React from 'react';
import classNames from 'classnames';

import styles from './FormFeedback.module.scss';

type Props = {
  children?: React.ReactNode;
  variant: 'success' | 'warning' | 'error';
};

const FormFeedback: React.FC<Props> = ({ children, variant = 'error' }: Props) => {
  const className = classNames(styles.formFeedback, {
    [styles.error]: variant === 'error',
    [styles.warning]: variant === 'warning',
    [styles.success]: variant === 'success',
  });
  return <div className={className}>{children}</div>;
};

export default FormFeedback;
