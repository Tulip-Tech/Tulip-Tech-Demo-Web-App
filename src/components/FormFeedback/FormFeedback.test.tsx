import React from 'react';
import { render } from '@testing-library/react';

import FormFeedback from './FormFeedback';

describe('<FormFeedback>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<FormFeedback variant="error">Form feedback</FormFeedback>);

    expect(container).toMatchSnapshot();
  });
});
