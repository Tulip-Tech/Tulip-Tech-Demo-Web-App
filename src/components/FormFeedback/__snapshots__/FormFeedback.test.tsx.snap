// Vitest Snapshot v1

exports[`<FormFeedback> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_formFeedback_99f247 _error_99f247"
  >
    Form feedback
  </div>
</div>
`;
